spec/unit/request_spec.rb

```diff
@@ -618,6 +618,22 @@
 
 
   describe "logging" do
+    it "logs custom user agent" do
+      log = RestClient.log = []
+      RestClient.user_agent = "Foo"
+      RestClient::Request.new(:method => :get, :url => 'http://url').log_request
+      RestClient.user_agent = nil
+      expect(log[0]).to eq %Q{RestClient.get "http://url", "Accept"=>"*/*", "User-Agent"=>"Foo"\n}
```

Comment:
This tests User-Agent the log _says_ is used, but ideally would test actually sent headers.
I think testing `req.processed_headers` is a good way to test.

Reply:
(and probably move the tests outside `describe "logging"` section)